### PR TITLE
Add is_set method to Event

### PR DESCRIPTION
**Pull request by ktbarrett**
Mirrors [`asyncio.Event`](https://docs.python.org/3/library/asyncio-sync.html#asyncio.Event.is_set). While `.fired` does exist, and is public, it is not documented.

### CODE REVIEW

**Comment by ktbarrett** [comment]
@eric-wieser Did I go overboard or underboard on the type annotations? I annotated everything but the special methods that have known output types, like `__repr__`. I put it through mypy and got it to not scream about anything.

**Comment by eric-wieser** [comment]
> @eric-wieser Did I go overboard or underboard on the type annotations?

If you're going to go that overboard, I'd prefer you do so in a standalone PR next time.

**Comment by eric-wieser** [comment]
Can you add a simple test too?

File: cocotb/triggers.py [hunk 1]
```diff
@@ -420,6 +420,10 @@ def clear(self):
         :meth:`~cocotb.triggers.Event.set` is called again."""
         self.fired = False
 
+    def is_set(self):
```
Comment (eric-wieser) [inline cocotb/triggers.py:423]:
```suggestion
    def is_set(self) -> bool:
```
while you're here, maybe

**Comment by eric-wieser** [comment]
Unless you deliberately want to save all the typing to be done at once, it's tempting to add the annotation while we're here. Feel free to commit through the github UI, I'll squash merge anyway.

**Pull request merged**